runtime/cpp/include/bpf-api.h

```diff
@@ -64,13 +66,22 @@ class wasm_bpf_program {
 
    public:
     int bpf_map_fd_by_name(const char *name);
-    int load_bpf_object(const void *obj_buf, size_t obj_buf_sz);
+    int load_bpf_object(const void* obj_buf,
+                        size_t obj_buf_sz);
     int attach_bpf_program(const char *name, const char *attach_target);
     int bpf_buffer_poll(wasm_exec_env_t exec_env, int fd, int32_t sample_func,
                         uint32_t ctx, void *buffer_data, size_t max_size,
                         int timeout_ms);
+    bpf_map* map_ptr_by_fd(int fd);
 };
 
+/// @brief A user data structure whose instance will be shared in a wasm
+/// runtime. It will store a map containing id->bpf_program and map fds opened
+/// by a bpf program
+/// Note that we need to remove fds opened by a bpf program when it's closed
+struct bpf_program_manager {
```

Comment:
using program manager = xxx

Reply:
I think we may add more global states shared in a single wasm program in the future, so I make this into a struct instead of a map. What do you think?

Reply:
Fixed now. I change `bpf_program_manager` to a type alias of the inside map